cpp/src/arrow/ipc/writer.cc

```diff
@@ -526,6 +526,14 @@ class RecordBatchSerializer {
     return VisitType(*array.indices());
   }
 
+  Status Visit(const RunLengthEncodedArray& array) {
+    --max_recursion_depth_;
+    RETURN_NOT_OK(VisitArray(*array.run_ends_array()));
+    RETURN_NOT_OK(VisitArray(*array.values_array()));
+    ++max_recursion_depth_;
+    return Status::OK();
+  }
```

Comment:
How are we handling the logical offset with the IPC?

Reply:
nvm, i see the usage of `logical_run_ends` and `logical_values`, but shouldn't this be using those instead?

Reply:
yes, you're right